sdk/core/azure-core/src/main/java/com/azure/core/exception/package-info.java

```diff
@@ -1,7 +1,31 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+
 /**
- * Package containing core exception classes.
+ * <p>This package contains the core exception classes used throughout the Azure SDKs.</p>
+ *
+ * <p>These exceptions are typically thrown in response to errors that occur when interacting with Azure services.
+ * For example, if a network request to an Azure service fails, an exception from this package might be thrown.
+ * The specific exception that is thrown depends on the nature of the error.</p>
+ *
+ * <p>Here are some of the key exceptions included in this package:</p>
+ * <ul>
+ *     <li>{@link com.azure.core.exception.AzureException}: The base class for all exceptions thrown by Azure SDKs.</li>
+ *
+ *     <li>{@link com.azure.core.exception.HttpRequestException}: Represents an exception thrown when an HTTP request
+ *     fails.</li>
+ *
+ *     <li>{@link com.azure.core.exception.HttpResponseException}: Represents an exception thrown when an unsuccessful
+ *     HTTP response is received from a service request.</li>
+ *
+ *     <li>{@link com.azure.core.exception.ResourceExistsException}: Represents an exception thrown when an HTTP request
+ *     attempts to create a resource that already exists.</li>
+ *
+ *     <li>{@link com.azure.core.exception.ResourceNotFoundException}: Represents an exception thrown when an HTTP
+ *     request attempts to access a resource that does not exist.</li>
+ * </ul>
+ *
+ * <p>Each exception provides useful information about the error, such as a message and, in some cases, the HTTP request or response that led to the exception.</p>
```

Comment:
```suggestion
 * <p>Some exceptions (noted in their documentation) include the HTTP request or response that led to the exception.</p>
```

Exceptions including a message isn't novel, but calling out that users can sometimes get the request/response is not a typical exception behavior.